hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,294 @@
+        final SignatureObject fileSignature = generateSignatureObject(streamDigest.digest(), sigKeyPair);
+        return SignatureFile.newBuilder().setFileSignature(fileSignature).setMetadataSignature(metadataSignature);
+    }
+
+    private static void generateSigRecordStreamFile(final File filePath, SignatureFile.Builder signatureFile)
+            throws IOException {
+        try (BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            output.write(RecordStreamType.getInstance().getSigFileHeader()[0]);
+            signatureFile.build().writeTo(output);
+        } catch (final IOException e) {
+            System.err.printf(
+                    "generateSigRecordStreamFile :: Failed to generate signature file for [%s] with exception : [%s]%n",
+                    filePath.getAbsolutePath(), e);
+            throw e;
+        }
+    }
+
+    private static void outputStreamDigest(int[] fileHeader, String recordFile) throws IOException {
+        try (final SerializableDataOutputStream dosMeta =
+                        new SerializableDataOutputStream(new HashingOutputStream(metadataStreamDigest));
+                final SerializableDataOutputStream dos = new SerializableDataOutputStream(
+                        new BufferedOutputStream(new HashingOutputStream(streamDigest)))) {
+
+            // parse record file
+            final Pair<Integer, Optional<RecordStreamFile>> recordResult =
+                    readMaybeCompressedRecordStreamFile(recordFile);
+            final var f = recordResult.getValue();
+            if (f.isPresent()) {
+                if (f.get().hasStartObjectRunningHash()) {
+                    final long blockNumber = f.get().getBlockNumber();
+                    final byte[] startRunningHash =
+                            f.get().getStartObjectRunningHash().getHash().toByteArray();
+                    final byte[] endRunningHash =
+                            f.get().getEndObjectRunningHash().getHash().toByteArray();
+                    final int version = recordResult.getKey();
+                    final byte[] serializedBytes = f.get().toByteArray();
+
+                    // update stream digest
+                    //                    System.out.printf("Writing file header [%s]%n", Arrays.toString(fileHeader));
+                    for (final int value : fileHeader) {
+                        dosMeta.writeInt(value);
+                    }
+                    //                    System.out.printf("Writing start running hash [%s]%n", hex(startRunningHash));
+                    dosMeta.write(startRunningHash);
+                    //                    System.out.printf("Writing end running hash [%s]%n", hex(endRunningHash));
+                    dosMeta.write(endRunningHash);
+                    //                    System.out.printf("Writing block number [%s]%n", blockNumber);
+                    dosMeta.writeLong(blockNumber);
```

Comment:
I suggest we keep the output log,
 it will be critical for debugging if something went wrong.
Use use log4j2 or standard output.

Reply:
updated to keep it as standard output.